app/content/views/short_link.py

```diff
@@ -0,0 +1,50 @@
+import re
+
+from django.shortcuts import get_object_or_404
+from rest_framework import status, viewsets
+from rest_framework.response import Response
+
+from app.common.permissions import IsMember, get_user_id
+from app.content.models import ShortLink, User
+from app.content.serializers import ShortLinkSerializer
+
+
+class ShortLinkViewSet(viewsets.ModelViewSet):
+    serializer_class = ShortLinkSerializer
+    queryset = ShortLink.objects.all()
+
+    def get_queryset(self):
+        if hasattr(self, "action") and self.action == "retrieve":
+            return self.queryset
+        else:
+            user = get_object_or_404(User, user_id=get_user_id(self.request))
+            return self.queryset.filter(user=user)
+
+    def get_permissions(self):
+        if self.request.method in ["POST"]:
+            self.permission_classes = [IsMember]
+        return super().get_permissions()
+
+    def create(self, request):
+        user = get_object_or_404(User, user_id=request.id)
+        serializer = ShortLinkSerializer(data=request.data)
+        regex = re.compile("(a|k|n|(om))/*.*", re.IGNORECASE)
+        if regex.match(request.data["name"]):
+            return Response(
+                {
+                    "detail": "Dette navnet er reservert. Navn som starter med 'om/', 'a/', 'k/' og 'n/' er ikke tilgjengelige."
+                },
+                status=status.HTTP_409_CONFLICT,
+            )
+        try:
+            serializer.save(user=user)
+            return Response(serializer.data, status=status.HTTP_200_OK)
+        except Exception:
```

Comment:
Vi prøver å unngå for brede except-setninger for å ikke sluke eventuelle feil/bugs. 


```suggestion
        except IntegrityError: 
```